app/models/collection_template.rb

```diff
@@ -54,16 +55,35 @@ def image_clean_to_formal_json
     json_params = HashWithIndifferentAccess.new(image_clean)
     pipeline_params = []
     json_params.each do |k, v|
-      if k.to_s == 'posterize'
-        pipeline_params.push(action: k, options:
-          { colors: v.to_i, method: json_params[:posterize_method] })
-      elsif k.to_s != 'posterize_method'
+      if k.to_s != 'posterize_method' && k.to_s != 'posterize'
         pipeline_params.push(action: k, options: { value: v.to_i })
       end
     end
     pipeline_params.to_json
   end
 
+  def posterize_params
+    json_params = HashWithIndifferentAccess.new(image_clean)
+    "-p '#{palette}' -m #{json_params['posterize_method']} " \
+      "#{json_params['posterize']}"
+  end
+
+  def palette
+    collection = Collection.find(collection_id)
+    collection.palette
```

Comment:
You could just use: `collection.palette` here

Also perhaps a more Ruby/Railsy way would be at the top of the file:

```
delegate : palette, to: : collection
```